### DISCUSSION TITLE
Question of auth and token

### DISCUSSION THREAD

**Discussion by Jalson1982** [invertase/react-native-firebase#4702] [Q&A] [answered]
Hello, 
I can not find how should handle the auth part correctly. Basically, my backend returns me a token, and with this token, I call auth().signInWithCustomToken(token). All works fine my users have access to chat and other things. Also, on signout I call auth.signout(). But the problem is if the user does not use the app for some time and when coming to the app we refresh token for the backend. But how to handle firebase part in this flow because user can not access chat, etc. Any idea how to handle refresh token or flow in this case?

**Comment by mikehardy** [comment, marked as answer]
Hi there! I have no experience in this area but I can say it will have nothing to do in essence with react-native-firebase, it will be a more generic way that firebase works. I encourage you to post on stackoverflow and tag it appropriately (with firebase etc) as I know the firebase team responds there, and optionally you could log an issue in one of the underlying related SDKs (firebase-js-sdk, firebase-android-sdk, firebase-ios-sdk) cross-linking your stackoverflow post in order to get official attention from the firebase team

When you do that please cross-link back to here so we can all learn from the answer

**Reply by Jalson1982** [reply]
Thanks, @mikehardy. I posted it on StackOverflow and hopefully will get an answer.
